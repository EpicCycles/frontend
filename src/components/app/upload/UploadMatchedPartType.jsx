import React from "react";
import {Icon} from "semantic-ui-react";
import * as PropTypes from "prop-types";

export const UploadMatchedPartType = props => (
    <div
        key={`match${props.matched.rowIndex}${props.matchIndex}`}
        className={(props.multiplesAllowed || (props.matchIndex === 0)) ? "rounded-auto" : "rounded-auto red"}
    >
        {props.matched.partTypeName}
        <Icon
            key={`matchDelete$${props.matched.rowIndex}${props.matchIndex}`}
            name="delete"
            onClick={() => props.undoMapping(props.matched.rowIndex)}
        />
    </div>
);
UploadMatchedPartType.propTypes = {
    matched: PropTypes.object.isRequired,
    matchIndex: PropTypes.number.isRequired,
    undoMapping: PropTypes.func.isRequired,
    multiplesAllowed: PropTypes.bool,
};