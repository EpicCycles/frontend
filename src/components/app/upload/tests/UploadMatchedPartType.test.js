import React from 'react';
import { Icon } from 'semantic-ui-react';
import toJson from 'enzyme-to-json';
import { UploadMatchedPartType } from '../UploadMatchedPartType';

const foundName = 'find me';
const rowMappings = [
  { rowIndex: 1, partType: 2, partTypeName: foundName },
  { rowIndex: 2, partTypeName: 'id 11', ignore: true },
  { rowIndex: 3, partTypeName: 'id 11' },
  { rowIndex: 4, partTypeName: 'id 111' },
  { rowIndex: 5, partTypeName: 'id 234' },
  { rowIndex: 11, partType: 2, partTypeName: foundName },
];

const undoMapping = jest.fn();
describe('UploadMatchedPartType', () => {
  test('it renders', () => {
    const component = shallow(
      <UploadMatchedPartType matchIndex={0} matched={rowMappings[0]} undoMapping={undoMapping} />,
    );
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('div.red')).toHaveLength(0);
  });
  test('it renders red when not first match', () => {
    const component = shallow(
      <UploadMatchedPartType matchIndex={1} matched={rowMappings[5]} undoMapping={undoMapping} />,
    );
    expect(toJson(component)).toMatchSnapshot();
    expect(component.find('div.red')).toHaveLength(1);
  });
  test('it calls undoMapping with the row present when the Icon is clicked', () => {
    const component = shallow(
      <UploadMatchedPartType matchIndex={1} matched={rowMappings[5]} undoMapping={undoMapping} />,
    );
    expect(component.find('div.red')).toHaveLength(1);
    expect(component.find(Icon)).toHaveLength(1);

    component
      .find(Icon)
      .at(0)
      .simulate('click');
    expect(undoMapping.mock.calls).toHaveLength(1);
    expect(undoMapping.mock.calls[0][0]).toEqual(11);
  });
});
