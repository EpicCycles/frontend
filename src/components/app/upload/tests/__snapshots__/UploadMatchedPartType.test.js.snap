// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadMatchedPartType it renders 1`] = `
<div
  className="rounded-auto"
  key="match10"
>
  find me
  <Icon
    as="i"
    key="matchDelete$10"
    name="delete"
    onClick={[Function]}
  />
</div>
`;

exports[`UploadMatchedPartType it renders red when not first match 1`] = `
<div
  className="rounded-auto red"
  key="match111"
>
  find me
  <Icon
    as="i"
    key="matchDelete$111"
    name="delete"
    onClick={[Function]}
  />
</div>
`;
